plugins/view-transitions/includes/settings.php

```diff
@@ -0,0 +1,401 @@
+			if ( is_string( $existing_key ) ) {
+				unset( $args[ $group . '-transition-names' ][ $existing_key ] );
+			}
+			$args[ $group . '-transition-names' ][ $options[ $option_name ] ] = $view_transition_name;
+		}
+	}
+
+	// phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+	$_wp_theme_features['view-transitions'] = $args;
+}
+
+/**
+ * Adds the settings sections and fields for the View Transitions configuration.
+ *
+ * @since 1.0.0
+ * @access private
+ */
+function plvt_add_setting_ui(): void {
+	add_settings_section(
+		'plvt_view_transitions',
+		__( 'View Transitions', 'view-transitions' ),
+		static function (): void {
+			?>
+			<p class="description">
+				<?php esc_html_e( 'This section allows you to control how view transitions are used to enhance the navigation user experience.', 'view-transitions' ); ?>
+			</p>
+			<?php
+		},
+		'reading',
+		array(
+			'before_section' => '<div id="view-transitions">',
+			'after_section'  => '</div>',
+		)
+	);
+
+	$reset_explanation = __( 'To reset, clear the field and save the changes.', 'view-transitions' );
+
+	$fields = array(
+		'default_transition_animation' => array(
+			'title'       => __( 'Default Transition Animation', 'view-transitions' ),
+			'description' => __( 'Choose the animation that is used for the default view transition type.', 'view-transitions' ),
+		),
+		'header_selector'              => array(
+			'title'       => __( 'Header Selector', 'view-transitions' ),
+			'description' => __( 'Provide the CSS selector to detect the global header element.', 'view-transitions' ) . ' ' . $reset_explanation,
+		),
+		'main_selector'                => array(
+			'title'       => __( 'Main Selector', 'view-transitions' ),
+			'description' => __( 'Provide the CSS selector to detect the global main element.', 'view-transitions' ) . ' ' . $reset_explanation,
+		),
+		'post_title_selector'          => array(
+			'title'       => __( 'Post Title Selector', 'view-transitions' ),
+			'description' => __( 'Provide the CSS selector to detect the post title element.', 'view-transitions' ) . ' ' . $reset_explanation,
+		),
+		'post_thumbnail_selector'      => array(
+			'title'       => __( 'Post Thumbnail Selector', 'view-transitions' ),
+			'description' => __( 'Provide the CSS selector to detect the post thumbnail element.', 'view-transitions' ) . ' ' . $reset_explanation,
+		),
+		'post_content_selector'        => array(
+			'title'       => __( 'Post Content Selector', 'view-transitions' ),
+			'description' => __( 'Provide the CSS selector to detect the post content element.', 'view-transitions' ) . ' ' . $reset_explanation,
+		),
+	);
+	foreach ( $fields as $slug => $args ) {
+		add_settings_field(
+			"plvt_view_transitions_{$slug}",
+			$args['title'],
+			'plvt_render_settings_field',
+			'reading',
+			'plvt_view_transitions',
+			array_merge(
+				array(
+					'field'     => $slug,
+					'label_for' => "plvt-view-transitions-field-{$slug}",
+				),
+				$args
+			)
```

Comment:
```suggestion
			array(
				'field'       => $slug,
				'label_for'   => "plvt-view-transitions-field-{$slug}",
				'description' => $args['description'] ?? '',
			)
```

We don’t need `$args['title']`; only `$args['description']` is required. So we can simplify the logic by leveraging that directly and removing the `array_merge` call.

Reply:
`$args['title']` is required by the PHPStan type annotation of `plvt_render_settings_field()`. It's technically not used at the moment, but it makes sense to include it in the data, since there can be fields that need to display the title within this callback (see e.g. the similar function in the Speculative Loading plugin).

Reply:
If warranted, the `title` field could be made optional making this change:

```diff
--- a/plugins/view-transitions/includes/settings.php
+++ b/plugins/view-transitions/includes/settings.php
@@ -295,11 +295,11 @@ function plvt_add_setting_ui(): void {
  * @since 1.0.0
  * @access private
  *
- * @param array{ field: non-empty-string, title: non-empty-string, description: string, label_for: non-empty-string } $args {
+ * @param array{ field: non-empty-string, title?: non-empty-string, description: string, label_for: non-empty-string } $args {
  *     Associative array of arguments.
  *
  *     @type string $field       The slug of the sub setting controlled by the field.
- *     @type string $title       The title for the field.
+ *     @type string $title       The title for the field. Not currently used.
  *     @type string $description Optional. A description to show for the field.
  *     @type string $label_for   ID to use for the field control.
  * }
```

Reply:
I don't think it should be made optional, conceptually it's certainly more important than a description.

The purpose of this function is a general field renderer for the plugin, and while currently it only implements text inputs and dropdowns (which have their label displayed "automatically" elsewhere via the Settings API), it might as well support checkbox lists and radio buttons (which would need to be in a `fieldset` with `legend`, i.e. the title would be needed here). So it makes sense to me to generally require all pieces of data that a field _might_ need.

Reply:
Gotcha. thanks for clearing that up!